tests/api_test.py

```diff
@@ -81,9 +82,47 @@ def test_resolve_nested_refs():
     Make sure we resolve nested refs gracefully and not get lost in
     the recursion. Also make sure we don't rely on dictionary order
     """
+    os.environ["PYTHONHASHSEED"] = str(1)
     with open('tests/sample_schemas/nested_defns/swagger.yaml') as swagger_spec:
         spec_dict = yaml.load(swagger_spec)
     spec = Spec.from_dict(spec_dict, '')
-    for i in range(1, 50):
-        os.environ["PYTHONHASHSEED"] = str(i)
-        resolve_refs(spec, spec_dict, ['/'], 'swagger', {})
+    resolve_refs(spec, spec_dict, ['/'], 'swagger', {})
+
+
+def test_external_refs():
+    """
+    resolve specs with external refs
+    """
+    with open("tests/sample_schemas/external_refs/swagger.json") as swagger_spec:
+        spec_dict = json.load(swagger_spec)
+    path = '{0}{1}'.format('file://', os.path.abspath('tests/sample_schemas/external_refs/swagger.json'))
+    spec = Spec.from_dict(spec_dict, path)
+    resolved_dict = resolve_refs(
+        spec=spec,
+        val=spec_dict,
+        json_path=['/'],
+        file_path='swagger.json',
+        defs_dict={},
+    )
+    correct_dict = {
+        'swagger': '2.0',
+        'paths': {
+            '/blah': {
+                'get': {
+                    'responses': {
+                        '200': {
+                            'schema': {
+                                '$ref': '#/definitions/:....A.json|'
```

Comment:
Hrm. Is there any way we can write this test without asserting the precise key that gets used here? Can we maybe *use* the spec dict returned by resolve_refs and try to use it to validate a request that exercises this bit of code?

As a general rule, higher level tests that assert the behavior users care about are better than lower level tests that assert implementation details...